addons/hr/models/hr_employee.py

```diff
@@ -351,9 +351,9 @@ def get_formview_id(self, access_uid=None):
         if access_uid:
             self_sudo = self.with_user(access_uid)
         else:
-            self_sudo = self
+            self_sudo = self.env.user
 
-        if self_sudo.user_has_groups('hr.group_hr_user'):
+        if self_sudo.has_group('hr.group_hr_user'):
```

Comment:
Would be better to just use the environnement here instead of changing `self_sudo`.
```suggestion
        if self_sudo.env.user.has_group('hr.group_hr_user'):
```